### ISSUE TITLE
Can I upgrade from Ventura to Sonoma and replace OC 0.9.9 efi?

### ISSUE THREAD

**Issue by wbews** [SchmockLord/Hackintosh-Intel-i9-10900k-Gigabyte-Z490-Vision-D#266]
Can I upgrade from Ventura to Sonoma and replace OC 0.9.9 efi? Have any special operations been canceled in the previous version?
If I install Sonoma from scratch, is it ok to use OC 0.9.9 efi?

**Comment by wbews** [comment]
The previous version was too complicated and I didn’t update it. Today I saw the update 0.9.9 and asked specifically about it.

**Comment by tokyoaudiofan** [comment]
I'm on 14.4 but I haven't tried 14.4.1

**Comment by garethsebshaw** [comment]
Hi, I’m struggling to get sonoma installed.
I found @SchmockLord ’s post but I think I just want understanding something or missing something. 

I’ve tried a regular install on a new drive, reboots, goes through the bios to boot page, select installer, then reboots again and again.

reboot back into Ventura.

then tried to patch it, clean drive, reinstall, reboot and I’d the same thing.

i checked in the bios and the settings are correct.
Checked in the EFI and they’re correct.
I did not add in the extra kext’s though. Is it simply that I need the kext? I’ve been trying the 14.4.1 installer.


Last night I switched to the 0.9.9 from 0.9.5 and on both of them my Bluetooth has stopped working. 

What am I missing? Thanks

**Comment by mueex** [comment]
May I ask you, which steps have you done? Like I told you, I am still on Sonoma 14.0, but in order to update from Ventura to Sonoma I just followed these steps:

To be repeated for every macOS Update:

- Disable SecureBoot via config.plist: SecureBootModel = Disabled and DmgLoading = Any
- Disable SecureBoot in BIOS
- Install the macOS Update
- Run the OCLP Patcher and start the "Post-install Root Patch".
- Enable SecureBoot via config.plist: SecureBootModel=j185f (iMac20,2) or j137 (iMacPro1,1) and DmgLoading = Any
- Enable SecureBoot in BIOS again

In my understanding this also means, that you have to change the settings mentioned above every time for performing of an Update. In addition you need to change the BIOS settings as well. Did you try it?

**Comment by mueex** [comment]
@garethsebshaw  Ok mate, you got me. I tried upgrading to Sonoma 14.4.1 on my own and... I failed. I was regretting this step of course, but there was no turning back. 

To update, I disabled SecureBootModel and just ran the Full installer (as I did previously months ago for Updating from Ventura to Sonoma 14.0). After installation, I observed kind of a kernel panic (I guess it was one), as it was definitely not a boot loop (I saw some lines appearing very quickly and then I got just a black screen + monitors went into standby mode). I started looking for a solution, which I found step by step. I discovered through my try-and-error journey that IOSkywalkFamily.kext was causing my black screen. I asked Google for help and found this [one](https://github.com/perez987/Broadcom-wifi-back-on-macOS-Sonoma-by-OCLP):
 - in short: since Sonoma 14.4 you have to replace IOSkywalkFamily.kext with version 1.1.0 from OLCP GitHub >> payloads >> Kexts >> Wifi >> [Link](https://github.com/dortania/OpenCore-Legacy-Patcher/tree/main/payloads/Kexts/Wifi)

After loading Sonoma 14.4.1, in the next step apply the OCLP patch in order to achieve a working wifi, change SecureBootModel back to your model's value, restart, reset NVRAM, and it should run smoothly for you as well.
And btw: since AMFIpass v1.4.0 you don't need amfi=0x80 or amfipassbeta as boot argument. I've found it on my journey as well. Just delete it. Good luck! 

![Screenshot 2024-04-30 at 23 20 22](https://github.com/SchmockLord/Hackintosh-Intel-i9-10900k-Gigabyte-Z490-Vision-D/assets/83857013/684fa3c2-cfa1-4ef0-8ac3-5f728374ab20)

**Comment by garethsebshaw** [comment]
sorry for the delay, i had been away. I'm going to have a look now and follow your instructions to see if i can do a fresh install.

**Comment by tokyoaudiofan** [comment]
What's the latest version of macOS that can be installed with the latest update? Can Sonoma 14.5 be installed?

**Comment by addisonduda** [comment]
@tokyoaudiofan I'm on Sonoma 14.5 with OC 1.0.0 using Chris's v35 release. The system is running great following the OCLP adjustments. I'm on the iMac 20,2 with iGPU display output version.

**Comment by tokyoaudiofan** [comment]
@addisonduda Thank you for the reply. I am trying to do a clean install using iMac 20,2 with iGPU for computing only (have a Radeon VII installed) but after cleaning the vram and starting the install I get a black screen and nothing happens. Followed all instructions carefully. 
Anyone knows what I am doing wrong?

**Comment by mueex** [comment]
@tokyoaudiofan Please try to debug the issue on your own. As we all use almost the same configuration and are able to carry out the update to Sonoma 14.5, the cause can only be a minor issue. Turn on the output (boot argument '-v') and check the lines before your screen goes black. In my case, when I was trying to update from Sonoma 14.0 to 14.4.1 as described in this thread above, it was all about IOSkywalkFamily.kext, which only needed to be updated. I'm not sure if Chris and I are using the same version since the file size is different, but I was also able to update to Sonoma 14.5 without any issues. I set SecureBootModel to Disabled and DmgLoading to Any in my config, performed the update, rebooted, changed the settings back to their previous values, reset NVRAM, and applied OCLP again to activate Wifi/BT. That's it.
